src/WebformHelper.php

```diff
@@ -0,0 +1,274 @@
+      ];
+    }
+
+    if ($this->currentUser->isAuthenticated()) {
+      /** @var \Drupal\user\Entity\User $apiUser */
+      $apiUser = $this->entityTypeManager->getStorage('user')->load($this->currentUser->id());
+      // Don't show API data links if current user is not included in
+      // (non-empty) list of allowed users.
+      if (!empty($allowedUsers) && !isset($allowedUsers[$apiUser->id()])) {
+        $apiUser = NULL;
+      }
+      $apiKey = $apiUser ? $apiUser->api_key->value : NULL;
+      if (!empty($apiKey)) {
+        $form['third_party_settings']['os2forms']['os2forms_rest_api']['api_info']['endpoints_test'] = [
+          '#type' => 'fieldset',
+          '#title' => $this->t('Test API endpoints'),
+
+          'links' => [],
+
+          'message' => [
+            '#markup' => $this->t('These are only for checking the API responses for user %user. <strong>Do not</strong> share these urls!', ['%user' => $apiUser->getAccountName()]),
+          ],
+        ];
+
+        $form['third_party_settings']['os2forms']['os2forms_rest_api']['api_info']['endpoints_test']['links']['#prefix'] = '<ol>';
+        $form['third_party_settings']['os2forms']['os2forms_rest_api']['api_info']['endpoints_test']['links']['#suffix'] = '</ol>';
+
+        foreach ($routes as $route) {
+          $parameters = [];
+
+          if ('rest.webform_rest_submit.POST' !== $route) {
+            $parameters['webform_id'] = $webform->id();
+          }
+          $uuidPlaceholder = '{uuid}';
+          if ($requireUuid($route)) {
+            $parameters['uuid'] = $uuidPlaceholder;
+          }
+          $parameters['api-key'] = $apiKey;
+
+          $url = Url::fromRoute($route, $parameters, ['absolute' => TRUE]);
+          $form['third_party_settings']['os2forms']['os2forms_rest_api']['api_info']['endpoints_test']['links'][$route] = [
+            '#type' => 'link',
+            '#title' => str_replace(urlencode($uuidPlaceholder), $uuidPlaceholder, $url->toString()),
+            '#url' => $url,
+            '#prefix' => '<li>',
+            '#suffix' => '</li>',
+          ];
+        }
+      }
+    }
+  }
+
+  /**
+   * Get webform by id or submission uuid.
+   *
+   * If submission uuid is specified (i.e. not null), the submission's webform's
+   * id must match the specified webform id.
+   *
+   * @return \Drupal\webform\WebformInterface|null
+   *   The webform if found.
+   */
+  public function getWebform(string $webformId, string $submissionUuid = NULL): ?WebformInterface {
+    if (NULL !== $submissionUuid) {
+      $storage = $this->entityTypeManager->getStorage('webform_submission');
+      $submissionIds = $storage
+        ->getQuery()
+        ->accessCheck(FALSE)
+        ->condition('uuid', $submissionUuid)
+        ->execute();
+      $submission = $storage->load(array_key_first($submissionIds));
+
+      if (NULL === $submission) {
+        return NULL;
+      }
+
+      assert($submission instanceof WebformSubmissionInterface);
```

Comment:
that looks like a leftover from something else

Reply:
It's to help the static code analysis. We could also use 

```php
/** @var WebformSubmissionInterface $submission */
```

but using an assertion actually makes sure that we have an instance of the expected class.